ocaml/src/tools/llio2.ml

```diff
@@ -55,6 +55,10 @@ module ReadBuffer = struct
         let res = a_from buf in
         assert (buffer_done buf);
         res
+    let maybe_from_buffer_no_assert a_from default buf =
```

Comment:
interesting fix, but don't we want to remove the assert everywhere?
it limits the extensibility of our serialisation ...
or maybe not remove it everywhere but at least evaluate all places that use maybe_from_buffer?